api/service/syncing/downloader/client.go

```diff
@@ -41,10 +41,12 @@ func (client *Client) Close() {
 }
 
 // GetBlockHashes gets block hashes from all the peers by calling grpc request.
-func (client *Client) GetBlockHashes(startHash []byte, size uint32) *pb.DownloaderResponse {
+func (client *Client) GetBlockHashes(startHash []byte, size uint32, ip, port string) *pb.DownloaderResponse {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 	request := &pb.DownloaderRequest{Type: pb.DownloaderRequest_HEADER, BlockHash: startHash, Size: size}
+	request.Ip = ip
+	request.Port = port
```

Comment:
where are those two fields being used?  I didn't see the IP/Port being used anyway?

Reply:
It's used in the CalculateResponse function in node/node_syncing.go